meilisearch-lib/src/tasks/task_store/store.rs

```diff
@@ -68,11 +68,42 @@ impl Store {
         let uids_task_ids = env.create_database(Some(UID_TASK_IDS))?;
         let tasks = env.create_database(Some(TASKS))?;
 
-        Ok(Self {
-            env,
-            uids_task_ids,
-            tasks,
-        })
+        // we are going to save all unfinished tasks for the `TaskStore`.
+        let mut unfinished_tasks: Vec<TaskId> = Vec::new();
+
+        let mut wtxn = env.write_txn()?;
+        let mut iter = tasks.rev_iter_mut(&mut wtxn)?;
+
+        while let Some(entry) = iter.next() {
+            let entry = entry?;
+            let (id, mut task): (BEU64, Task) = entry;
+
+            // Since all tasks are ordered, we can stop iterating on all tasks when we encounter our first non-finished task.
+            if task.is_finished() {
+                break;
+            }
+
+            // we only keep the first state. It’s supposed to be a `Created` state.
+            task.events.drain(1..);
+            unfinished_tasks.push(id.get());
+
+            // the id stayed unchanged so this is a safe operation.
+            unsafe {
+                iter.put_current(&id, &task)?;
+            }
+        }
+
+        drop(iter);
+        wtxn.commit()?;
```

Comment:
i think this could be its own function

Reply:
uhm i even think that rather than returning the pending tasks when created, we could have `TaskStore` call that method and initialize it's queue itself.